int32 state
bool mask_odom
